src/Symfony/Component/PropertyInfo/Extractor/ReflectionExtractor.php

```diff
@@ -63,6 +63,9 @@ public function getProperties($class, array $context = array())
         foreach ($reflectionClass->getMethods(\ReflectionMethod::IS_PUBLIC) as $reflectionMethod) {
             $propertyName = $this->getPropertyName($reflectionMethod->name);
             if ($propertyName) {
+                if (!isset($properties[$propertyName])) {
+                    $propertyName = lcfirst($propertyName);
```

Comment:
Not really:

```
> /[A-Z]{2,}$/.test('SIRET')
true //don't lowercase first char
> /[A-Z]{2,}$/.test('Siret')
false //do lowercase
> /[A-Z]{2,}$/.test('SIret')
false //same
> /[A-Z]{2,}/.test('SIret')
true //with your regex it'd become SIret, but it should be sIret
```

Regex tests that the full property name is uppercased. If it's not, I think we should lower the first character.